frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -170,11 +161,34 @@ const CreateSnapshotForm = withHandlePromise<SnapshotResourceProps>((props) => {
   }, [namespace, pvcName]);
 
   const [data, loaded, loadError] = useK8sWatchResource<PersistentVolumeClaimKind[]>(resourceWatch);
+  const scList: StorageClassResourceKind[] = scObjListLoaded
+    ? (scObjList.items as StorageClassResourceKind[])
+    : [];
+  const snapshotClassFilter = useCallback(
+    (snapshotClass: VolumeSnapshotClassKind) => scObj?.provisioner.includes(snapshotClass?.driver),
+    [scObj],
+  );
+  const vscList: VolumeSnapshotClassKind[] = vscLoaded
+    ? (vscObj.items as VolumeSnapshotClassKind[])
+    : [];
+  const getDefaultItem = useCallback(() => {
+    const filteredVSC: VolumeSnapshotClassKind[] = vscList.filter(snapshotClassFilter);
+    const defaultFilteredVSC: VolumeSnapshotClassKind[] = filteredVSC.filter(
+      isDefaultSnapshotClass,
+    );
+    const defaultItem: string = getName(defaultFilteredVSC?.[0]) || getName(filteredVSC?.[0]);
+
+    return defaultItem;
+  }, [vscList, snapshotClassFilter]);
 
   React.useEffect(() => {
     const currentPVC = data.find((pvc) => pvc.metadata.name === pvcName);
     setPVCObj(currentPVC);
-  }, [data, pvcName, namespace, loadError]);
+    setSCName(currentPVC?.spec?.storageClassName);
+    const currentSC = scList.find((sc) => sc.metadata.name === scName);
+    setSCObj(currentSC);
+    setSnapshotClassName(getDefaultItem());
```

Comment:
why is it required inside this if it's already part of handlePVCChange method(https://github.com/openshift/console/pull/7050/files#diff-157896dc2763ee1b7cd120a4460780b6f75fd6542fa8b45df1dae4c4c52bfac7R201)?

Reply:
i have given some thought about it. i added that so that we always recalculate the latest storage class object in case the lists change and use it for the snapshotClassFilter. I have moved this calculation to handlePVCName and introduced a new filterWrapper to get the latest value for the snapshotClassFilter